src/nunit.analyzers/DisposeFieldsAndPropertiesInTearDown/DisposeFieldsAndPropertiesInTearDownAnalyzer.cs

```diff
@@ -555,19 +554,60 @@ private static void DisposedIn(Parameters parameters, HashSet<string> disposals,
                 return memberAccessExpression.Name.Identifier.Text;
             }
 
-            // considering cast to IDisposable, e.g. in case of explicit interface implementation of IDisposable.Dispose()
-            else if (expression is ParenthesizedExpressionSyntax parenthesizedExpression &&
-                     parenthesizedExpression.Expression is CastExpressionSyntax castExpression &&
-                     castExpression.Expression is IdentifierNameSyntax castIdentifierName &&
-                     castExpression.Type is IdentifierNameSyntax typeIdentifierName &&
-                     typeIdentifierName.Identifier.Text.Equals("IDisposable", StringComparison.Ordinal))
+            // considering cast to I(Async)Disposable, e.g. in case of explicit interface implementation of IDisposable.Dispose()
+            // or in case of 'as IDisposable' or 'as IAsyncDisposable'
+            // or in case of 'is IDisposable' or 'is IAsyncDisposable'
```

Comment:
Mostly a comment - as I think the code below looks good for cast and `as`-pattern, but I don't think `is` is handled here (only in lines 481-486) or am I overlooking something?

Reply:
No you are right, I added the comment until my nunit tests proofed me wrong that it was done when checking `if` statement.